RMT symbols and musical buzzer example bugfix for ESP32

dsheets opened the pull request:
First, I introduce a newtype in the RMT module to remove the exposure of `rmt_item32_t` in the interface.

Then, I fix the `rmt_musical_buzzer` example to work on ESP32 (and maybe others) which is not supported by the ESP-IDF repo example. In the process, the example has now become a nascent 1-bit synthesizer.

ivmarkov commented:
I just merged [this](https://github.com/esp-rs/esp-idf-hal/commit/e87abce3b29be5a5ac76cf0688ee08c5babdeaa1). I think it goes against your idea of not exposing the raw ESP IDF type directly. Can you rebase and review, if additional changes are needed?

ivmarkov commented:
@dsheets ping?

dsheets commented:
Hi, sorry for the delay. I might have some time to take a look at this next week.

dsheets commented:
Ok, @ivmarkov, I've rebased and removed `Pulse::into_rmt_item` in favor of `Symbol::new` or `Symbol::update`. I also added a `Signal` implementation for a single `Symbol`. Rust currently has some difficulty with zero-cost commutation of type constructors and newtype inner type projections so no trait impl has been given for `[Symbol]` and the trait for `[rmt_item_t]` has been retained.

Using the present implementation I've now successfully listened to Ode to Joy from an ESP32 running it.

The exposure of `rmt_item_t` is still present but perhaps hurts slightly less now. When the rmt driver is migrated off the legacy implementation, this IDF type will change name (to `rmt_symbol_t`, iirc) so it seems valuable to me to have the newtype wrapper (also for other reasons including function association, etc).

Let me know if you need anything further -- I'm happy to help but can't guarantee timelines.
